### ISSUE TITLE
REST path with querystring returns a 404

### ISSUE THREAD

**Issue by FrancescoFontana** [mailjet/mailjet-apiv3-nodejs#104]
Hi,
it seems that when the path is built, the querystring is added after a **/**, leading to a 404.

https://github.com/mailjet/mailjet-apiv3-nodejs/blob/2166e4ba457d3c136bb06351728c057649aeb01d/mailjet-client.js#L251

I suppose the slash before the question mark should not be there.

Thanks!
Francesco

**Comment by svengau** [comment]
Hi @adamyanliev 

I think this bug is really serious, because it breaks the API usage on my side. I can't pass a query anymore.
The line highlighted by @FrancescoFontana is not new (2017) so it means something has changed in your API in production.

Please, could you check it ? Thanks

**Comment by svengau** [comment]
@FrancescoFontana as a workaround, you can directly pass the query string in the url.

Example:

```
mailjet
      .get('template?OwnerType=user&Limit=100&Purposes=transactional', { version: 'v3' })
```

**Comment by Boelensman1** [comment]
Alternatively you can monkey-patch mailjet untill they patch it on their end.

```
const mailjet = require('node-mailjet')

const qs = require('querystring')
const _path = require('path')
mailjet.prototype.path = function(resource, sub, params, options) {
  const url = options && 'url' in options ? options.url : this.config.url
  const api_version =
    options && 'version' in options ? options.version : this.config.version

  var base = _path.join(api_version, sub)
  if (Object.keys(params).length === 0) {
    return _path.join(url, base + '/' + resource)
  }

  var q = qs.stringify(params).replace(/%2B/g, '+')
  return _path.join(url, base + '/' + resource + '?' + q)
  // last line was edited to remove extra /
  // see https://github.com/mailjet/mailjet-apiv3-nodejs/issues/104
}
```

**Comment by HugoPoi** [comment]
Ok I encountered the issue like a month ago and the filters had work before last month so the API HAVE changed some how.
According to https://www.npmjs.com/package/node-mailjet#pass-parameters-as-well-as-a-callback
This should work
```javascript
  const Mailjet = require('node-mailjet').connect(process.env.MJ_APIKEY_PUBLIC, process.env.MJ_APIKEY_PRIVATE, {version: 'v3'});
    return Mailjet.get('template').request({
      Purposes: 'transactional',
    }).then(response => {
      return response.body.Data;
    });
```
It throw `Unsuccessful: 404 The resource with the specified ID you are trying to reach does not exist.`

**Comment by Marc-Pierre-Barbier** [comment]
i created a pull request https://github.com/mailjet/mailjet-apiv3-nodejs/pull/161

**Comment by scroll17** [comment]
Hey @FrancescoFontana @svengau @Boelensman1 @HugoPoi @Marc-Pierre-Barbier 

About problem: 
I tried to reproduce the same problem with other resources but they work is correctly and that mean what resource '/template' is processed in some unique way that is different from other resources. 
But still slash before query string is extra.

This problem was fixed in release `3.3.9` and this release already available.
